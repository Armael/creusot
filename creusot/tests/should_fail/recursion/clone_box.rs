#![allow(unused)]
extern crate creusot_contracts;
use creusot_contracts::*;

struct S;

impl Clone for S {
    fn clone(&self) -> Self {
        let _ = Box::new(S).clone();
        S
    }
}
