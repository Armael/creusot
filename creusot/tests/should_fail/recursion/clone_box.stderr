error: Mutually recursive functions: when calling <S as creusot_contracts::Clone>::clone...
    --> clone_box.rs:8:5
     |
8    |     fn clone(&self) -> Self {
     |     ^^^^^^^^^^^^^^^^^^^^^^^
     |
note: then <S as creusot_contracts::Clone>::clone calls <std::boxed::Box<T, A> as creusot_contracts::Clone>::clone...
    --> clone_box.rs:9:29
     |
9    |         let _ = Box::new(S).clone();
     |                             ^^^^^^^
note: finally <std::boxed::Box<T, A> as creusot_contracts::Clone>::clone calls <S as creusot_contracts::Clone>::clone.
    --> /home/arnaud/.rustup/toolchains/nightly-2024-01-31-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1301:5
     |
1301 |     fn clone(&self) -> Self {
     |     ^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

